### PR TITLE
Merge unit-tests into master

### PR DESCRIPTION
**Pull request by ddrobner**
Wrote some basic unit tests, take a look at them and tell me what should be fixed. To be honest it's my first time doing this, so I'm not 100% sure what's right/wrong or good/bad so take a look at them and let me know.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -70,7 +70,8 @@ dependencies {
     nativeZip wpi.deps.vendor.jni(wpi.platforms.roborio)
     nativeDesktopZip wpi.deps.vendor.jni(wpi.platforms.desktop)
 
-    testImplementation 'junit:junit:4.12'
+    implementation 'junit:junit:4.12'
```
Comment (andrewazores) [inline build.gradle]: This should also be `testImplementation`

File: build.gradle [hunk 2]
```diff
@@ -70,7 +70,8 @@ dependencies {
...
+    testImplementation 'org.mockito:mockito-core:+'
```
Comment (andrewazores) [inline build.gradle]: Specify a version number

File: src/test/java/frc/robot/subsystems/ball/intake/IntakeImplTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    @Test
+    public void testRetract() {
+        // Retract the intake
+        intakeSubsystem.retract();
+
+        // Check behaviour
+        assertEquals(false, intakeSubsystem.getDeployed());
+    }
```
Comment (andrewazores) [inline src/test/java/frc/robot/subsystems/ball/intake/IntakeImplTest.java:57]: You can also use `Mockito.verify()` to check that calling a method on the object to test results in it calling some other methods on mocks. ex. `intakeSubsystem.retract(); intakeSubsystem.task(); Mockito.verify(intakeSolenoid).set(whateverDirection)`.

Comment (andrewazores) [reply src/test/java/frc/robot/subsystems/ball/intake/IntakeImplTest.java:57]: You can also use this same technique to test more complex behaviours - you could for example retract, run task, verify solenoid action, then extend, run task, and verify the second solenoid action. Not a particularly interesting example for this simple subsystem, but still a useful tool.

**Review by andrewazores — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (ddrobner) [reply build.gradle]: Thought I changed it back after I removed the class I wrote to run tests in a single instance. Will change that.

File: src/test/java/frc/robot/subsystems/ball/intake/IntakeImplTest.java (hunk 3 above)
Comment (ddrobner) [reply src/test/java/frc/robot/subsystems/ball/intake/IntakeImplTest.java:57]: I added some more to the tests using verify(). Right now it doesn't make sense to do any more with this but I'll definitely end up writing more unit tests soon, so now that I know how to do this I can do some more complex behaviour, and structure the tests differently as well.

**Review by andrewazores — APPROVED** [review]

**Pull request merged**